File: ccif/src/java/other/org/himalay/ccif/ApplicationInfoReqest.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
// 
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:32:55 EST 2016")

public  class ApplicationInfoReqest extends   APDUBaseFactory.APDUBase { //Concrete type is ApplicationInfoReqest




    // members variables
    // header
    public APDUHeader header ;
    // length
    public short length ;
    // displayRow
    public int displayRow ;
    // displayColumns
    public int displayColumns ;
    // verticalScrolling
    public short verticalScrolling ;
    // horizontalScrolling
    public short horizontalScrolling ;
    // dataEntrySupport
    public short dataEntrySupport ;
    // HTMLSupport
    public short HTMLSupport ;
    // htmlCapabilities
    public HTMLCapabilities htmlCapabilities ;

        public ApplicationInfoReqest () // throws Exception
    {
            init();
    }
    
    private void init()
    {
        // Initialize header
        header = new APDUHeader();
        // Initialize length
        
        // Initialize displayRow
        
        // Initialize displayColumns
        
        // Initialize verticalScrolling
        
        // Initialize horizontalScrolling
        
        // Initialize dataEntrySupport
        
        // Initialize HTMLSupport
        
        // Initialize htmlCapabilities
        //Conditional, will be initialized during read
    }

    public int readNoHeader(DataInputStream istream) throws IOException 
    {
    
  preRead();
    int retVal= 0;
                // read length
        {length=(short)(istream.readUnsignedByte()); retVal+=1;}
        // read displayRow
        {displayRow=istream.readUnsignedShort(); retVal+=2;}
        // read displayColumns
        {displayColumns=istream.readUnsignedShort(); retVal+=2;}
        // read verticalScrolling
        {verticalScrolling=(short)(istream.readUnsignedByte()); retVal+=1;}
        // read horizontalScrolling
        {horizontalScrolling=(short)(istream.readUnsignedByte()); retVal+=1;}
        // read dataEntrySupport
        {dataEntrySupport=(short)(istream.readUnsignedByte()); retVal+=1;}
        // read HTMLSupport
        {HTMLSupport=(short)(istream.readUnsignedByte()); retVal+=1;}
        // read htmlCapabilities
        if (HTMLSupport == 1){ htmlCapabilities = new HTMLCapabilities();retVal += htmlCapabilities.read(istream);        }

postRead();
        return retVal;
        }
   
    public int read(DataInputStream istream) throws IOException 
    {
    preRead();
        int retVal= 0;

                     // read header
        retVal += header.read(istream);
        // read length
        {length=(short)(istream.readUnsignedByte()); retVal+=1;}
        // read displayRow
        {displayRow=istream.readUnsignedShort(); retVal+=2;}
        // read displayColumns
        {displayColumns=istream.readUnsignedShort(); retVal+=2;}
        // read verticalScrolling
        {verticalScrolling=(short)(istream.readUnsignedByte()); retVal+=1;}
        // read horizontalScrolling
        {horizontalScrolling=(short)(istream.readUnsignedByte()); retVal+=1;}
        // read dataEntrySupport
        {dataEntrySupport=(short)(istream.readUnsignedByte()); retVal+=1;}
        // read HTMLSupport
        {HTMLSupport=(short)(istream.readUnsignedByte()); retVal+=1;}
        // read htmlCapabilities
        if (HTMLSupport == 1){ htmlCapabilities = new HTMLCapabilities();retVal += htmlCapabilities.read(istream);        }

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException 
    {
    preWrite();
        int retVal= 0;

        { /** fix dependent sizes for header **/  }
        
        
        
        
        
        
        
        { /** fix dependent sizes for htmlCapabilities **/  }
    
                // write header
        if (header!=null)retVal +=header.write(ostream);
        // write length
        ostream.writeByte(length); retVal +=1;
        // write displayRow
        ostream.writeShort(displayRow); retVal +=2;
        // write displayColumns
        ostream.writeShort(displayColumns); retVal +=2;
        // write verticalScrolling
        ostream.writeByte(verticalScrolling); retVal +=1;
        // write horizontalScrolling
        ostream.writeByte(horizontalScrolling); retVal +=1;
        // write dataEntrySupport
        ostream.writeByte(dataEntrySupport); retVal +=1;
        // write HTMLSupport
        ostream.writeByte(HTMLSupport); retVal +=1;
        // write htmlCapabilities
        if (htmlCapabilities!=null)retVal +=htmlCapabilities.write(ostream);
postWrite();
        return retVal;
    }
    
    public int dump(DumpContext dc) throws IOException 
    {
        dc.indent();dc.getPs().print("ApplicationInfoReqest\n");
    dc.increaseIndent();
        int retVal= 0;
        // write header
        if ( header != null ) {dc.indent();dc.getPs().println("header") ;retVal +=header.dump(dc);}
        // write length
        dc.indent();dc.getPs().println("length="+length+"(0x"+ Integer.toHexString(length) + ")" );
        // write displayRow
        dc.indent();dc.getPs().println("displayRow="+displayRow+"(0x"+ Integer.toHexString(displayRow) + ")" );
        // write displayColumns
        dc.indent();dc.getPs().println("displayColumns="+displayColumns+"(0x"+ Integer.toHexString(displayColumns) + ")" );
        // write verticalScrolling
        dc.indent();dc.getPs().println("verticalScrolling="+verticalScrolling+"(0x"+ Integer.toHexString(verticalScrolling) + ")" );
        // write horizontalScrolling
        dc.indent();dc.getPs().println("horizontalScrolling="+horizontalScrolling+"(0x"+ Integer.toHexString(horizontalScrolling) + ")" );
        // write dataEntrySupport
        dc.indent();dc.getPs().println("dataEntrySupport="+dataEntrySupport+"(0x"+ Integer.toHexString(dataEntrySupport) + ")" );
        // write HTMLSupport
        dc.indent();dc.getPs().println("HTMLSupport="+HTMLSupport+"(0x"+ Integer.toHexString(HTMLSupport) + ")" );
        // write htmlCapabilities
        if ( htmlCapabilities != null ) {dc.indent();dc.getPs().println("htmlCapabilities") ;retVal +=htmlCapabilities.dump(dc);}
dc.decreaseIndent();
        return retVal;
    }


        // Getter for header
    //public APDUHeader getHeader()
    //{
    //    return header ;
    //}

    
    // Setter for header
    //public void setHeader(APDUHeader val)
    //{
    //    this.header= val;
    //}
    // Getter for length
    //public short getLength()
    //{
    //    return length ;
    //}

    
    // Setter for length
    //public void setLength(short val)
    //{
    //    this.length= val;
    //}
    // Getter for displayRow
    //public int getDisplayRow()
    //{
    //    return displayRow ;
    //}

    
    // Setter for displayRow
    //public void setDisplayRow(int val)
    //{
    //    this.displayRow= val;
    //}
    // Getter for displayColumns
    //public int getDisplayColumns()
    //{
    //    return displayColumns ;
    //}

    
    // Setter for displayColumns
    //public void setDisplayColumns(int val)
    //{
    //    this.displayColumns= val;
    //}
    // Getter for verticalScrolling
    //public short getVerticalScrolling()
    //{
    //    return verticalScrolling ;
    //}

    
    // Setter for verticalScrolling
    //public void setVerticalScrolling(short val)
    //{
    //    this.verticalScrolling= val;
    //}
    // Getter for horizontalScrolling
    //public short getHorizontalScrolling()
    //{
    //    return horizontalScrolling ;
    //}

    
    // Setter for horizontalScrolling
    //public void setHorizontalScrolling(short val)
    //{
    //    this.horizontalScrolling= val;
    //}
    // Getter for dataEntrySupport
    //public short getDataEntrySupport()
    //{
    //    return dataEntrySupport ;
    //}

    
    // Setter for dataEntrySupport
    //public void setDataEntrySupport(short val)
    //{
    //    this.dataEntrySupport= val;
    //}
    // Getter for HTMLSupport
    //public short getHTMLSupport()
    //{
    //    return HTMLSupport ;
    //}

    
    // Setter for HTMLSupport
    //public void setHTMLSupport(short val)
    //{
    //    this.HTMLSupport= val;
    //}
    // Getter for htmlCapabilities
    //public HTMLCapabilities getHtmlCapabilities()
    //{
    //    return htmlCapabilities ;
    //}

    
    // Setter for htmlCapabilities
    //public void setHtmlCapabilities(HTMLCapabilities val)
    //{
    //    this.htmlCapabilities= val;
    //}


    public int getSize() throws IOException
    {
       DataOutputStream dos= new DataOutputStream(new NullStream());
       return this.write(dos);
    }
    
        
     public void setHeader(APDUHeader header)
    {
         this.header= header;   
    }
    
    public APDUHeader getHeader()
    {
        return this.header;
    }

}

// End of code

File: libccif/src/bp/ApplicationInfoReqest.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:12 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "ApplicationInfoReqest.h"

#ifdef org_himalay_ccif_ApplicationInfoReqest__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
ApplicationInfoReqest::ApplicationInfoReqest() {
    references= 0;


    // header
    org_himalay_ccif_APDUHeader__NEW( header);
    // length
    length= (ui8)0;
    // displayRow
    displayRow= (ui16)0;
    // displayColumns
    displayColumns= (ui16)0;
    // verticalScrolling
    verticalScrolling= (ui8)0;
    // horizontalScrolling
    horizontalScrolling= (ui8)0;
    // dataEntrySupport
    dataEntrySupport= (ui8)0;
    // HTMLSupport
    HTMLSupport= (ui8)0;
    // htmlCapabilities
    org_himalay_ccif_HTMLCapabilities__INIT(htmlCapabilities)
}
// Destructor
ApplicationInfoReqest::~ApplicationInfoReqest() {
    // header
    org_himalay_ccif_APDUHeader__DELETE( header);
    // length

    // displayRow

    // displayColumns

    // verticalScrolling

    // horizontalScrolling

    // dataEntrySupport

    // HTMLSupport

    // htmlCapabilities
    org_himalay_ccif_HTMLCapabilities__DELETE( htmlCapabilities);
}

int ApplicationInfoReqest::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // length
    {
        length=(istream.readUI8());
        retVal+=1;
    }
    // displayRow
    {
        displayRow=istream.readUI16();
        retVal+=2;
    }
    // displayColumns
    {
        displayColumns=istream.readUI16();
        retVal+=2;
    }
    // verticalScrolling
    {
        verticalScrolling=(istream.readUI8());
        retVal+=1;
    }
    // horizontalScrolling
    {
        horizontalScrolling=(istream.readUI8());
        retVal+=1;
    }
    // dataEntrySupport
    {
        dataEntrySupport=(istream.readUI8());
        retVal+=1;
    }
    // HTMLSupport
    {
        HTMLSupport=(istream.readUI8());
        retVal+=1;
    }
    // htmlCapabilities
    if (HTMLSupport == 1) {
        org_himalay_ccif_HTMLCapabilities__NEW( htmlCapabilities);
        retVal += htmlCapabilities->read(istream);
    }
    return retVal;
}

int ApplicationInfoReqest::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read header
    retVal += header->read(istream);
    // read length
    {
        length=(istream.readUI8());
        retVal+=1;
    }
    // read displayRow
    {
        displayRow=istream.readUI16();
        retVal+=2;
    }
    // read displayColumns
    {
        displayColumns=istream.readUI16();
        retVal+=2;
    }
    // read verticalScrolling
    {
        verticalScrolling=(istream.readUI8());
        retVal+=1;
    }
    // read horizontalScrolling
    {
        horizontalScrolling=(istream.readUI8());
        retVal+=1;
    }
    // read dataEntrySupport
    {
        dataEntrySupport=(istream.readUI8());
        retVal+=1;
    }
    // read HTMLSupport
    {
        HTMLSupport=(istream.readUI8());
        retVal+=1;
    }
    // read htmlCapabilities
    if (HTMLSupport == 1) {
        org_himalay_ccif_HTMLCapabilities__NEW( htmlCapabilities);
        retVal += htmlCapabilities->read(istream);
    }

    return retVal;
}

int ApplicationInfoReqest::write(DataOutputStream& ostream) { // throws IOException
    {   /** fix dependent sizes for header **/
    }







    {   /** fix dependent sizes for htmlCapabilities **/
    }

    int retVal= 0;
    // write header
    if ( header != NULL ) {
        retVal +=header->write(ostream);
    }
    // write length
    ostream.writeUI8(length);
    retVal +=1;
    // write displayRow
    ostream.writeUI16(displayRow);
    retVal +=2;
    // write displayColumns
    ostream.writeUI16(displayColumns);
    retVal +=2;
    // write verticalScrolling
    ostream.writeUI8(verticalScrolling);
    retVal +=1;
    // write horizontalScrolling
    ostream.writeUI8(horizontalScrolling);
    retVal +=1;
    // write dataEntrySupport
    ostream.writeUI8(dataEntrySupport);
    retVal +=1;
    // write HTMLSupport
    ostream.writeUI8(HTMLSupport);
    retVal +=1;
    // write htmlCapabilities
    if ( htmlCapabilities != NULL ) {
        retVal +=htmlCapabilities->write(ostream);
    }

    return retVal;
}

int ApplicationInfoReqest::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("ApplicationInfoReqest\n");
    dc.increaseIndent();
    int retVal= 0;
    // write header
    if ( header != NULL ) {
        dc.indent();
        dc.getPs().println("header") ;
        retVal +=header->dump(dc);
    }
    // write length
    dc.indent();
    dc.getPs().print("length=");
    dc.getPs().println((long)length);
    // write displayRow
    dc.indent();
    dc.getPs().print("displayRow=");
    dc.getPs().println((long)displayRow);
    // write displayColumns
    dc.indent();
    dc.getPs().print("displayColumns=");
    dc.getPs().println((long)displayColumns);
    // write verticalScrolling
    dc.indent();
    dc.getPs().print("verticalScrolling=");
    dc.getPs().println((long)verticalScrolling);
    // write horizontalScrolling
    dc.indent();
    dc.getPs().print("horizontalScrolling=");
    dc.getPs().println((long)horizontalScrolling);
    // write dataEntrySupport
    dc.indent();
    dc.getPs().print("dataEntrySupport=");
    dc.getPs().println((long)dataEntrySupport);
    // write HTMLSupport
    dc.indent();
    dc.getPs().print("HTMLSupport=");
    dc.getPs().println((long)HTMLSupport);
    // write htmlCapabilities
    if ( htmlCapabilities != NULL ) {
        dc.indent();
        dc.getPs().println("htmlCapabilities") ;
        retVal +=htmlCapabilities->dump(dc);
    }
    dc.decreaseIndent();
    return retVal;
}


// Getter for header
//org_himalay_ccif_APDUHeader__PTR_TYPE ApplicationInfoReqest::getHeader()
//{
//return this->header;
//}

// Setter for header
//void ApplicationInfoReqest::setHeader(org_himalay_ccif_APDUHeader__PTR_TYPE val)
//{
//this->header= val;
//}

// Getter for length
//ui8 ApplicationInfoReqest::getLength()
//{
//return this->length;
//}

// Setter for length
//void ApplicationInfoReqest::setLength(ui8 val)
//{
//this->length= val;
//}

// Getter for displayRow
//ui16 ApplicationInfoReqest::getDisplayRow()
//{
//return this->displayRow;
//}

// Setter for displayRow
//void ApplicationInfoReqest::setDisplayRow(ui16 val)
//{
//this->displayRow= val;
//}

// Getter for displayColumns
//ui16 ApplicationInfoReqest::getDisplayColumns()
//{
//return this->displayColumns;
//}

// Setter for displayColumns
//void ApplicationInfoReqest::setDisplayColumns(ui16 val)
//{
//this->displayColumns= val;
//}

// Getter for verticalScrolling
//ui8 ApplicationInfoReqest::getVerticalScrolling()
//{
//return this->verticalScrolling;
//}

// Setter for verticalScrolling
//void ApplicationInfoReqest::setVerticalScrolling(ui8 val)
//{
//this->verticalScrolling= val;
//}

// Getter for horizontalScrolling
//ui8 ApplicationInfoReqest::getHorizontalScrolling()
//{
//return this->horizontalScrolling;
//}

// Setter for horizontalScrolling
//void ApplicationInfoReqest::setHorizontalScrolling(ui8 val)
//{
//this->horizontalScrolling= val;
//}

// Getter for dataEntrySupport
//ui8 ApplicationInfoReqest::getDataEntrySupport()
//{
//return this->dataEntrySupport;
//}

// Setter for dataEntrySupport
//void ApplicationInfoReqest::setDataEntrySupport(ui8 val)
//{
//this->dataEntrySupport= val;
//}

// Getter for HTMLSupport
//ui8 ApplicationInfoReqest::getHTMLSupport()
//{
//return this->HTMLSupport;
//}

// Setter for HTMLSupport
//void ApplicationInfoReqest::setHTMLSupport(ui8 val)
//{
//this->HTMLSupport= val;
//}

// Getter for htmlCapabilities
//org_himalay_ccif_HTMLCapabilities__PTR_TYPE ApplicationInfoReqest::getHtmlCapabilities()
//{
//return this->htmlCapabilities;
//}

// Setter for htmlCapabilities
//void ApplicationInfoReqest::setHtmlCapabilities(org_himalay_ccif_HTMLCapabilities__PTR_TYPE val)
//{
//this->htmlCapabilities= val;
//}

int ApplicationInfoReqest::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

void ApplicationInfoReqest::setHeader(org_himalay_ccif_APDUHeader__PTR_TYPE header)
{
    this->header= header;
}

org_himalay_ccif_APDUHeader__PTR_TYPE ApplicationInfoReqest::getHeader()
{
    return this->header;
}

#ifdef org_himalay_ccif_ApplicationInfoReqest__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::ApplicationInfoReqest* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::ApplicationInfoReqest* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code